addons/point_of_sale/models/res_config_settings.py

```diff
@@ -1,21 +1,316 @@
 # -*- coding: utf-8 -*-
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
+    """
+    NOTES
+    1. Fields with 'pos' attributes are removed from the vals before super call to `create`.
+       Values of these fields are written to `pos_config_id` record after the super call.
+       So, if we want to make an atomic write to some set of fields, we need to redefine those
+       fields to have compute, readonly=False, store=True, and pos='<pos.config.field.name>'.
+    2. During `creation` of this record, each related field is written to the source record
+       *one after the other*, so constraints on the source record that are based on multiple
+       fields might not work properly. However, only the *modified* related fields are written
+       to the source field. But the identification of modified fields happen during the super
+       call, not before `create` is called. Because of this, vals contains a lot of field before
+       super call, then the number of fields is reduced after.
+    """
     _inherit = 'res.config.settings'
 
+    def _valid_field_parameter(self, field, name):
+        """Introduce 'pos' attribute to allow atomic write for the destination pos.config record."""
+        return name == 'pos' or super()._valid_field_parameter(field, name)
+
+    def _default_pos_config(self):
+        # Default to the last modified pos.config.
+        return self.env['pos.config'].search([('company_id', '=', self.env.company.id)], order='write_date desc', limit=1)
+
+    pos_config_id = fields.Many2one('pos.config', string="Point of Sale", default=lambda self: self._default_pos_config())
     sale_tax_id = fields.Many2one('account.tax', string="Default Sale Tax", related='company_id.account_sale_tax_id', readonly=False)
     module_pos_mercury = fields.Boolean(string="Vantiv Payment Terminal", help="The transactions are processed by Vantiv. Set your Vantiv credentials on the related payment method.")
     module_pos_adyen = fields.Boolean(string="Adyen Payment Terminal", help="The transactions are processed by Adyen. Set your Adyen credentials on the related payment method.")
     module_pos_six = fields.Boolean(string="Six Payment Terminal", help="The transactions are processed by Six. Set the IP address of the terminal on the related payment method.")
     update_stock_quantities = fields.Selection(related="company_id.point_of_sale_update_stock_quantities", readonly=False)
     account_default_pos_receivable_account_id = fields.Many2one(string='Default Account Receivable (PoS)', related='company_id.account_default_pos_receivable_account_id', readonly=False)
+    is_default_pricelist_displayed = fields.Boolean(compute="_compute_is_default_pricelist_displayed")
+    is_cashdrawer_displayed = fields.Boolean(compute='_compute_is_cashdrawer_displayed')
```

Comment:
it does not seems to be really used as it is not in a view or something else. Moreover it is used as depends in "_compute_pos_iface_cashdrawer" which is probably not a good idea to use a compute fields as depends of another.

It should be possible to only declare "_compute_pos_iface_cashdrawer" and put the logic and depends from "is_cashdrawer_displayed" in it. (if we need to take a function out for an override, we can. But not a compute field to do so)